gui/wxpython/tools/update_menudata.py

```diff
@@ -140,31 +140,30 @@ def main(argv=None):
         print(sys.stderr, __doc__, file=sys.stderr)
         return 1
 
-    nuldev = open(os.devnull, "w+")
     grass.info("Step 1: running make...")
-    grass.call(["make"], stderr=nuldev)
-    grass.info("Step 2: parsing modules...")
-    modules = {}
-    modules = parseModules()
-    grass.info("Step 3: reading menu data...")
-    data = LayerManagerMenuData()
-    grass.info("Step 4: updating menu data...")
-    updateData(data, modules)
-
-    if printDiff:
-        tempFile = tempfile.NamedTemporaryFile()
-        grass.info("Step 5: diff menu data...")
-        writeData(data, tempFile.name)
-
-        grass.call(
-            ["diff", "-u", os.path.join("xml", "menudata.xml"), tempFile.name],
-            stderr=nuldev,
-        )
-    else:
-        grass.info("Step 5: writing menu data (menudata.xml)...")
-        writeData(data)
+    with open(os.devnull, "w+") as nuldev:
+        grass.call(["make"], stderr=nuldev)
```

Comment:
This whole thing is indetend just because `subprocess.DEVNULL`. Did you consider adding it?

Reply:
Sorry, what do you mean by "adding it"?

Reply:
Ok, I think I understood now. Looking at https://github.com/python/cpython/blob/e119526edface001ad7d7f70249a123c8a122d71/Lib/subprocess.py, I understand that subprocess would manage closing by itself the `subprocess.DEVNULL` (its value is -3). So I applied it.

Reply:
There are other usages of os.devnull in the codebase, and some are used for Popen (or a subclass of Popen, with grass.script.core.call()). Should it be changed at some point too?